### PR TITLE
feat: Add new Modal Component

### LINKED ISSUE #620: [Component] Modal/Dialog
doc - https://www.figma.com/file/0QchRdipAVuvVoDfTjLrgQ/Spark-Specs-and-Handoff?type=design&node-id=6255-36810&mode=design&t=5bV4wOcTAAfswfxw-0

### PR DESCRIPTION
**Pull request by soulcramer**
## 📋 Changes

Add the Modal component that adapts it's content based on the form factor used

## 🤔 Context

close #620

## 📸 Screenshots

image

## 🗒️ Other info

The configurator needs to be done, also we can't screenshot test this component as paparazzi doesn't render content from another window yet

### CODE REVIEW

File: spark/src/main/kotlin/com/adevinta/spark/tokens/WindowSizeClass.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    error("No WindowSizeClass available")
+}
+
+@ExperimentalMaterial3WindowSizeClassApi
+@Composable
+internal fun calculateWindowSizeClass(): WindowSizeClass {
+    // Observe view configuration changes and recalculate the size class on each change. We can't
+    // use Activity#onConfigurationChanged as this will sometimes fail to be called on different
+    // API levels, hence why this function needs to be @Composable so we can observe the
+    // ComposeView's configuration changes.
```
Comment (kazaky) [inline spark/src/main/kotlin/com/adevinta/spark/tokens/WindowSizeClass.kt:51]: It's better to use /* */ multiline comment in this case

File: spark/src/main/kotlin/com/adevinta/spark/tokens/Layout.kt [hunk 2]
```diff
@@ -27,82 +27,54 @@ import androidx.compose.foundation.layout.Box
...
 
-    // TODO: Use M3 WindowHeightSizeClass instead of our custom size ranges
-
-    @OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
     public val windowSize: WindowSizeClass
-        @Composable get() {
-            // Observe view configuration changes and recalculate the size class on each change. We can't
-            // use Activity#onConfigurationChanged as this will sometimes fail to be called on different
-            // API levels, hence why this function needs to be @Composable so we can observe the
-            // ComposeView's configuration changes.
```
Comment (kazaky) [inline spark/src/main/kotlin/com/adevinta/spark/tokens/Layout.kt:70]: Same here,
kindly use multiline comment /* */

**Review by kazaky — DISMISSED** [review]
Looks good,
Kindly check the 2 small comments

File: spark/src/main/kotlin/com/adevinta/spark/tokens/WindowSizeClass.kt (hunk 1 above)
Comment (SimonMarquis) [reply spark/src/main/kotlin/com/adevinta/spark/tokens/WindowSizeClass.kt:51]: Shouldn't it directly go as a javadoc comment instead?

**Pull request merged**